bin/node/cli/benches/block_production.rs

```diff
@@ -0,0 +1,218 @@
+			block_construction: sc_client_api::ExecutionStrategy::NativeWhenPossible,
+			offchain_worker: sc_client_api::ExecutionStrategy::NativeWhenPossible,
+			other: sc_client_api::ExecutionStrategy::NativeWhenPossible,
+		},
+		rpc_http: None,
+		rpc_ws: None,
+		rpc_ipc: None,
+		rpc_ws_max_connections: None,
+		rpc_cors: None,
+		rpc_methods: Default::default(),
+		rpc_max_payload: None,
+		ws_max_out_buffer_capacity: None,
+		prometheus_config: None,
+		telemetry_endpoints: None,
+		default_heap_pages: None,
+		offchain_worker: OffchainWorkerConfig { enabled: true, indexing_enabled: false },
+		force_authoring: false,
+		disable_grandpa: false,
+		dev_key_seed: Some(Sr25519Keyring::Alice.to_seed()),
+		tracing_targets: None,
+		tracing_receiver: Default::default(),
+		max_runtime_instances: 8,
+		announce_block: true,
+		base_path: Some(base_path),
+		informant_output_format: Default::default(),
+		wasm_runtime_overrides: None,
+	};
+
+	node_cli::service::new_full_base(config, |_, _| ()).expect("creating a full node doesn't fail")
+}
+
+fn extrinsic_transfer(client: &FullClient, nonce: u32) -> OpaqueExtrinsic {
+	create_extrinsic(
+		client,
+		Sr25519Keyring::Alice.pair(),
+		BalancesCall::transfer {
+			dest: Sr25519Keyring::Bob.to_account_id().into(),
+			value: 1 * DOLLARS,
+		},
+		Some(nonce),
+	)
+	.into()
+}
+
+fn extrinsic_set_time(now: u64) -> OpaqueExtrinsic {
+	node_runtime::UncheckedExtrinsic {
+		signature: None,
+		function: node_runtime::Call::Timestamp(pallet_timestamp::Call::set { now }),
+	}
+	.into()
+}
+
+fn import_block(
+	mut client: &FullClient,
+	built: BuiltBlock<
+		node_primitives::Block,
+		<FullClient as sp_api::CallApiAt<node_primitives::Block>>::StateBackend,
+	>,
+) {
+	let mut params = BlockImportParams::new(BlockOrigin::File, built.block.header);
+	params.state_action =
+		StateAction::ApplyChanges(sc_consensus::StorageChanges::Changes(built.storage_changes));
+	params.fork_choice = Some(ForkChoiceStrategy::LongestChain);
+	futures::executor::block_on(client.import_block(params, Default::default()))
+		.expect("importing a block doesn't fail");
+}
+
+fn block_production(c: &mut Criterion) {
+	sp_tracing::try_init_simple();
+
+	let runtime = tokio::runtime::Runtime::new().expect("creating tokio runtime doesn't fail; qed");
+	let tokio_handle = runtime.handle().clone();
+
+	let node = new_node(tokio_handle.clone());
+	let client = &*node.client;
+
+	// Buliding the very first block is around ~30x slower than any subsequent one,
+	// so let's make sure it's built and imported before we benchmark anything.
+	let mut block_builder = client.new_block(Default::default()).unwrap();
+	block_builder.push(extrinsic_set_time(1)).unwrap();
+	import_block(client, block_builder.build().unwrap());
+
+	let max_transfer_count = {
+		let mut transfer_count = 0;
+		let mut block_builder = client.new_block(Default::default()).unwrap();
+		block_builder.push(extrinsic_set_time(1 + 1500)).unwrap();
```

Comment:
```suggestion
		block_builder.push(extrinsic_set_time(1 + MILLISECS_PER_BLOCK)).unwrap();
```

Reply:
Ah, right, you're right that I should use a proper constant for this, however this actually can't be `MILLISECS_PER_BLOCK`. (:

Since BABE isn't actually running and we're using the `BlockBuilder` directly to author the blocks the current timeslot in BABE doesn't actually progress, so the code fails with a `Timestamp slot must match 'CurrentSlot'` assertion if we actually set the time to the next block.

So what I'm doing here is incrementing the timestamp only by the `MinimumPeriod` (which is 1500, half of the 
`SLOT_DURATION`/`MILLISECS_PER_BLOCK`), so BABE doesn't complain since it's still within the first timeslot.

If we had to actually import more blocks than the first one this would obviously not work, but we don't.

Is this actually a problem here?

Reply:
No should be fine.